src/coreclr/vm/readytoruninfo.cpp

```diff
@@ -1771,5 +1789,161 @@ ModuleBase* CreateNativeManifestModule(LoaderAllocator* pLoaderAllocator, IMDInt
+{
+    uint32_t rid = RidFromToken(input);
+    if ((rid > TypeCount) || (rid == 0))
+    {
+        return pImport->GetNestedClassProps(input, pEnclosingType);
+    }
+
+    *pEnclosingType = TokenFromRid((&TypeCount)[rid], mdtTypeDef);
+
+    if (*pEnclosingType == mdTypeDefNil)
+    {
+        return CLDB_E_RECORD_NOTFOUND;
+    }
+    return  S_OK;
+}
+
+ReadyToRunTypeGenericInfo ReadyToRun_TypeGenericInfoMap::GetTypeGenericInfo(mdTypeDef input, bool *foundResult) const
+{
+    uint32_t rid = RidFromToken(input);
+    if ((rid > TypeCount) || (rid == 0))
+    {
+        *foundResult = false;
+        return (ReadyToRunTypeGenericInfo)0;
+    }
+    uint32_t index = rid - 1;
+    uint8_t *pTypeGenericInfo = ((uint8_t*)&TypeCount) + sizeof(uint32_t);
+    uint8_t entry = pTypeGenericInfo[index / 2];
+    if (!(index & 1))
+    {
+        entry >>= 4;
+    }
+    *foundResult = true;
+    return static_cast<ReadyToRunTypeGenericInfo>(entry);
+}
+
+bool ReadyToRun_TypeGenericInfoMap::IsGeneric(mdTypeDef input, IMDInternalImport* pImport) const
+{
+    bool foundResult;
+    ReadyToRunTypeGenericInfo typeGenericInfo = GetTypeGenericInfo(input, &foundResult);
+    if (!foundResult)
+    {
+        HENUMInternalHolder hEnumTyPars(pImport);
+        hEnumTyPars.EnumInit(mdtGenericParam, input);
+        return (pImport->EnumGetCount(&hEnumTyPars) != 0);
+    }
+    return !!((uint8_t)typeGenericInfo & (uint8_t)ReadyToRunTypeGenericInfo::GenericCountMask);
+}
+
+HRESULT ReadyToRun_TypeGenericInfoMap::IsGenericNoThrow(mdTypeDef input, bool *pIsGeneric, IMDInternalImport* pImport) const
+{
+    bool foundResult;
+    bool result;
+    HRESULT hr;
+    ReadyToRunTypeGenericInfo typeGenericInfo = GetTypeGenericInfo(input, &foundResult);
+    if (!foundResult)
+    {
+        HENUMInternalHolder hEnumTyPars(pImport);
+        IfFailRet(hEnumTyPars.EnumInitNoThrow(mdtGenericParam, input));
+        result = (pImport->EnumGetCount(&hEnumTyPars) != 0);
+    }
+    else
+        result = !!((uint8_t)typeGenericInfo & (uint8_t)ReadyToRunTypeGenericInfo::GenericCountMask);
+
+    *pIsGeneric = result;
+    return S_OK;
+}
+
+uint32_t ReadyToRun_TypeGenericInfoMap::GetGenericArgumentCount(mdTypeDef input, IMDInternalImport* pImport) const
+{
+    bool foundResult;
+    ReadyToRunTypeGenericInfo typeGenericInfo = GetTypeGenericInfo(input, &foundResult);
+    uint32_t count = ((uint8_t)typeGenericInfo & (uint8_t)ReadyToRunTypeGenericInfo::GenericCountMask);
+    if (count > 2)
+        foundResult = false;
+    
+    if (!foundResult)
+    {
+        HENUMInternalHolder hEnumTyPars(pImport);
+        hEnumTyPars.EnumInit(mdtGenericParam, input);
+        return pImport->EnumGetCount(&hEnumTyPars);
+    }
+    return count;
+}
+
+HRESULT ReadyToRun_TypeGenericInfoMap::GetGenericArgumentCountNoThrow(mdTypeDef input, uint32_t *pCount, IMDInternalImport* pImport) const
```

Comment:
Nit - there seems to be tons of duplication between the pairs of methods with / without the <code>NoThrow</code> prefix. Could we somehow consolidate them by making one call the other and just slightly tweak the outputs?